core/src/main/java/io/questdb/griffin/engine/table/ShowCreateTableRecordCursorFactory.java

```diff
@@ -0,0 +1,273 @@
+                    }
+                    if (withO3MaxLag) {
+                        if (withMaxUncommittedRows) {
+                            sink.put(", ");
+                        }
+                        sink.put("o3MaxLag=").put(table.getO3MaxLag()).put("us");
+                    }
+                }
+
+                // IN VOLUME OTHER_VOLUME
+                if (table.getIsSoftLink()) {
+                    if (withRequired) {
+                        sink.put(',');
+                    }
+                    sink.put(" IN VOLUME ");
+
+                    FilesFacade ff = config.getFilesFacade();
+                    try (Path softLinkPath = new Path().of(config.getRoot()).concat(table.getDirectoryName())) {
+                        try (Path otherVolumePath = new Path()) {
+                            ff.readLink(softLinkPath, otherVolumePath);
+                            otherVolumePath.trimTo(otherVolumePath.size()
+                                    - table.getDirectoryName().length()  // look for directory
+                                    - 1 // get rid of trailing slash
+                            );
+                            CharSequence alias = config.getVolumeDefinitions().resolvePath(otherVolumePath.asAsciiCharSequence());
+
+                            if (alias == null) {
+                                throw CairoException.nonCritical().put("could not find volume alias for table [table=").put(tableToken).put(']');
+                            } else {
+                                sink.put(alias);
+                            }
+                        }
+                    }
+                }
+
+                // DEDUP UPSERT(key1, key2)
+                if (table.getIsDedup()) {
+                    sink.put('\n');
+                    sink.put("DEDUP UPSERT KEYS(");
+                    for (int i = 0, n = (int) table.getColumnCount(); i < n; i++) {
+                        final CairoColumn column = table.getColumnQuiet(i);
+                        if (column.getIsDedupKey()) {
+                            sink.put(column.getName());
+                            sink.put(',');
+                        }
+                    }
+                    // drop the last comma
+                    sink.clear(sink.size() - 1);
+                    sink.put(')');
+                }
+                sink.put(';');
+                hasRun = true;
+                return true;
+            }
+            return false;
+        }
+
+        public ShowCreateTableCursor of(SqlExecutionContext executionContext, TableToken tableToken, int tokenPosition) throws SqlException {
+            this.tableToken = tableToken;
+            this.executionContext = executionContext;
+            try (MetadataCacheReader metadataRO = executionContext.getCairoEngine().getMetadataCache().readLock()) {
+                this.table = metadataRO.getTable(tableToken);
```

Comment:
table can be deleted concurrently with this code, this is a repeated mistake (we have been fixing issues like this around metadata). TableToken cannot be presumed to exist. It has to be verified. We have to decide either this function is bound to table name, or token. We must not mix two together. For example. If function is bind to table name, we assume it is ok for function to return different DDL in case table was dropped and re-created before between factory creation and cursor invocation. If function is bound to token, then in the same example function must error out with table ref out of date exception. The confusing aspect of the current implementation is that is uses tokens around the code, but behaves as if it is bound to table name and not token.